### ISSUE TITLE
`mikeio.read_pfs` fail to parse PFS file when File name contains comma

### ISSUE THREAD

**Issue by zhangr4** [DHI/mikeio#763]
**Describe the bug**
PFS file, File name is delimited by a bar: |filename|. 

When File name contains comma, `mikeio.read_pfs` fail to parse PFS file with error message

```
yaml.scanner.ScannerError: while scanning for the next token
found character '|' that cannot start any token
```

**To Reproduce**
Here is a simplified example PFS file, create a PFS file and use `mikeio.read_pfs` read the file
```
[MIKE_ZERO_PLOT_COMPOSER]
   [ACTIVE_X_PLOT]
      [PLOT_PROPERTIES]
         [Files]
            Number = 1
            [DataFile]
               DataFileName = |C:\the path,to\test.dfs2|
            EndSect  // DataFile

         EndSect  // Files

      EndSect  // PLOT_PROPERTIES

   EndSect  // ACTIVE_X_PLOT

EndSect  // MIKE_ZERO_PLOT_COMPOSER
```

**Expected behavior**
- `mikeio.read_pfs` method could correctly parse file name. 
- Could the error message more clearly? Presently, the error seems throw from yaml scanner and indicate that '|' is not a start token, it takes me more time to locate the error comes from ',' instead of '|'.

**Screenshots**
![image](https://github.com/user-attachments/assets/b0cd8378-7422-4d8e-8cf6-51601a8206ce)

**System information:**
 - Python version: python 3.13.0
 - MIKE IO version: mikeio 2.2.0

**Comment by jsmariegaard** [comment]
I guess this is the same as #475

**Comment by zhangr4** [comment]
oh, yes, it is the same, shall I just close this issue? And btw, is there any plan to fix it?

**Comment by jsmariegaard** [comment]
@ecomodeller, is this now fixed with #782 ?

**Comment by ecomodeller** [comment]
No, I forgot to take | into account. It only works for strings in single quotes.

![Image](https://github.com/user-attachments/assets/ffc9ea05-a268-4450-b22e-a0623626ef1b)
